compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +515,48 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+hlintSettings :: FilePath -> IO ([Classify], Hint)
+hlintSettings hlintDataDir = do
+  -- `findSettings` ends up calling `readFilesConfig` which in turn
+  -- calls `readFileConfigYaml` which finally calls `decodeFileEither`
+  -- from the `yaml` library.  Annoyingly that function catches async
+  -- exceptions and in particular, it ends up catching `ThreadKilled`.
+  -- So, we have to mask to stop it from doing that.
+  (_, classify, hints) <- mask $ \unmask ->
+    findSettings (unmask . readSettingsFile (Just hlintDataDir)) Nothing
+  return (classify, hints)
+
+getHlintSettingsRule :: Maybe FilePath -> Rules ()
+getHlintSettingsRule dir =
+    defineNoFile $ \GetHlintSettings -> do
+      case dir of
+        Just dir -> liftIO $ hlintSettings dir
```

Comment:
Probably makes sense to pull `liftIO` out of these branches.

Reply:
✔️